idaes/power_generation/carbon_capture/mea_solvent_system/unit_models/kettlereboiler.py

```diff
@@ -0,0 +1,367 @@
+                            block=self.liquid_phase, doc='inlet Port')
+        self.add_outlet_port(name="liquid_outlet",
+                             block=self.liquid_phase, doc='outlet Port')
+
+        # Vapor Phase : No inlet port
+        self.add_outlet_port(name="vapor_outlet",
+                             block=self.vapor_phase, doc='outlet Port')
+        # ======================================================================
+        # Add performace equation method
+        self._make_params()
+        self._make_performance_method()
+
+    def _make_params(self):
+
+        # Internal variables for custom mass and energy balance
+
+        self.y = Var(self.flowsheet().time,
+                     self.config.vapor_side.property_package.component_list,
+                     initialize=0.5,
+                     units=None,
+                     doc="Vapor component composition")
+
+        self.vf = Var(self.flowsheet().time,
+                      initialize=0.5,
+                      units=None,
+                      doc="Vapor phase fraction")
+
+        self.FV = Var(self.flowsheet().time,
+                      initialize=1,
+                      units=units.kmol,
+                      doc="Vapor exit flow")
+
+        self.heat_duty = Var(self.flowsheet().time,
+                             initialize=1,
+                             units=units.kW,
+                             doc="Reboiler heat duty")
+
+    def _make_performance_method(self):
+
+        # =====================================================================
+        # Custom Material Balance
+        @self.Constraint(self.flowsheet().time,
+                         self.config.liquid_side.property_package.component_list,
+                         doc="Reboiler Material Balance")
+        def reb_material_balance(blk, t, i):
+            if i == 'MEA':
+                return blk.liquid_phase.properties_in[t].mole_frac_comp[i] ==\
+                    (1 - blk.vf[t]) * blk.liquid_phase.properties_out[t].mole_frac_comp[i]
+            else:
+                return blk.liquid_phase.properties_in[t].mole_frac_comp[i] ==\
+                    (1 - blk.vf[t]) * blk.liquid_phase.properties_out[t].mole_frac_comp[i] + blk.vf[t] * blk.y[t, i]
+
+        @self.Constraint(self.flowsheet().time,
+                         doc="Vapor exit flow")
+        def vap_exit_flow(blk, t):
+            return blk.FV[t] == blk.vf[t] *\
+                blk.liquid_phase.properties_in[t].flow_mol*1e-3
+
+        @self.Constraint(self.flowsheet().time,
+                         doc="Liquid exit flow")
+        def liq_exit_flow(blk, t):
+            return blk.liquid_phase.properties_out[t].flow_mol ==\
+                   blk.liquid_phase.properties_in[t].flow_mol - blk.FV[t]*1e3
+
+        # =====================================================================
+        # Custom Energy Balance
+
+        def rule_heat_vaporization(blk, t):
+            return 40650
```

Comment:
Why is this hard coded?

Reply:
Sure, this can easily be called from the property package, I will update that. My focus is to provide a flowsheet that is tractable for the desired optimization problem by the end of next month for the MEA application. If it means writing a letter on why this should be specific, we need to do that. Moreso every flowsheet is for a specific application.